### ISSUE TITLE
Error: Can't verify or update TAP tables: All columns must have defined unique column order between 1 and N:ivoa.tap_tables.params

### ISSUE THREAD

**Issue by gsleap** [csiro-rds/casda_vo_tools#14]
Hi,
I'm migrating my test instance from v1.8 to the latest v1.11, and I'm hitting a road block:
"Can't verify or update TAP tables: All columns must have defined unique column order between 1 and N:ivoa.tap_tables.params" whenever I "Update" my config via the web interface.

My tap db is empty except for my own tables (i.e. I had recreated the TAP db and CASDA VO tools has not created any of it's "tap_" tables yet).

So with a fresh install, and with a very basic config file, I get this error. CASDA VO tools manages to create the "tap_schemas" table, but with no rows. No other tables get created. It looks like it failed trying to create the "tap_tables" table (especially based on the error message).

Any ideas?  See below for my config (db connection info has been removed). The column orders are definitely unique and in sequence. But as above I think it's an issue about creating the "tap_" tables and it hasn't looked at mine yet.

```
!Configuration
endPoints: 
   TAP: 
      options: !Map
         tap.job.name.prefix: vo_req_
         tap.data.access.url: http://example.org:8000/casda_vo_tools/tap/jobResults/
         tap.max.running.jobs: 4
         tap.max.records: 20000000
         tap.async.base.url: /tap
         tap.async.description: UWS 
         tap.async.job.list.name: async
         tap.language.name: ADQL
         tap.language.version: 2.0
         tap.language.description: ADQL 2.0
         tap.output.format.mime: text/xml
         tap.output.format.alias: votable
         tap.retention.period.default: 432000
         tap.retention.period.hard: 432000
         tap.execution.duration.default: 360000
         tap.execution.duration.hard: 360000
         tap.sync.timeout: 20
         tap.async.timeout: 36000
         tap.output.limit.hard: 20000000
         log.timezone: UTC
         tap.votable.xsl: 
         tap.examples.url: 
      tables: 
      - ivoa.obscore_vw
   SCS: 
      options: !Map
         scs.output.format.mime: text/xml
         scs.outputFormat.alias: votable
         scs.max.radius: 10
         scs.max.records: 20000
         scs.test.ra: 100.0
         scs.test.dec: -30.0
         scs.test.schema: ivoa
         scs.test.catalog: obscore_vw
         scs.test.verbose: 3
         scs.test.extras: 
      tables: 
      - ivoa.obscore_vw
options: !Map
   application.base.url: http://example.org:8000/example
   application.message: Example VO Tools
   log.timezone: UTC
   connection.url: jdbc:postgresql://tap_db/db_name
   connection.username: secret
   connection.password: secret
   connection.driverClassName: org.postgresql.Driver
   build.environment: Production
   auth.trusted.ip: 
   auth.trusted.userId: 
   stylesheet.address: https://data.csiro.au/dap/resources-2.6.6/css/style.css
   logo.url: https://data.csiro.au/dap/resources-2.6.6/images/csiro_logo.png
   default.db.schema: ivoa
   metadata.instrument: 'Sample Instrument|Instrument from which data originated '
   metadata.server: 'http://tap.example.org:8000/casda_vo_tools/tap|URL of the VO endpoint used '
   metadata.serviceShortName: 'Example TAP|Short name for the TAP service '
   metadata.serviceTitle: 'Example TAP service|TAP service title '
   metadata.identifier: ivo://tap.example.org:8000/casda_vo_tools/TAP|Unique resource registry identifier
   metadata.servicePublisher: The TAP team|Publisher for the TAP service
   metadata.furtherInformation: http://www.example.org|Link to further information on the data and
      usage of the service
   metadata.contactPerson: 'email@email.org|Who to contact about this service '
   metadata.copyright: Creative Commons Attribution 4.0 International Licence|Example;apos;s Data license
   ssap.table: 
   ssap.outputLimit.hard: 20000000
   ssap.max.records: 1000
   ssap.metadata.response: 
schemas: 
   example: !au.csiro.casda.votools.config.SchemaConfig
      options: !Map
         description: Example specific tables
   ivoa: !au.csiro.casda.votools.config.SchemaConfig
      options: !Map
         description: IVOA standard tables
tables: 
   ivoa.obscore_vw: 
      columns: !Map
         obs_id: !Column
            options: !Map
               column_order: 1
               size: 8
               db_column_name: obs_id
               scs_verbosity: 1
               description: Observation ID
               utype: DataID.observationID
               principal: 1
               indexed: 0
               std: 1
            type: character varying
         dataproduct_type: !Column
            options: !Map
               column_order: 2
               size: 50
               db_column_name: dataproduct_type
               scs_verbosity: 3
               description: Logical data product type
               utype: ObsDataset.dataProductType
               principal: 0
               indexed: 0
               std: 1
            type: text
         dataproduct_subtype: !Column
            options: !Map
               column_order: 3
               size: 20
               db_column_name: dataproduct_subtype
               scs_verbosity: 3
               description: Logical data product subtype
               utype: ObsDataset.dataProductSubtype
               principal: 0
               indexed: 0
               std: 1
            type: text
         calib_level: !Column
            options: !Map
               column_order: 4
               size: 4
               db_column_name: calib_level
               scs_verbosity: 3
               description: Calibration level (0, 1, 2, 3, 4)
               utype: ObsDataset.calibLevel
               principal: 0
               indexed: 0
               std: 1
            type: smallint
         target_name: !Column
            options: !Map
               column_order: 5
               size: 255
               db_column_name: target_name
               scs_verbosity: 3
               description: Astronomical object observed if any
               utype: Target.name
               principal: 0
               indexed: 0
               std: 1
            type: text
         obs_title: !Column
            options: !Map
               column_order: 6
               size: 255
               db_column_name: obs_title
               scs_verbosity: 3
               description: Title of the data observation
               utype: DataID.title
               principal: 0
               indexed: 0
               std: 1
            type: text
         obs_collection: !Column
            options: !Map
               column_order: 7
               size: 255
               db_column_name: obs_collection
               scs_verbosity: 3
               description: Name of the data collection
               utype: DataID.collection
               principal: 0
               indexed: 0
               std: 1
            type: text
         obs_creator_name: !Column
            options: !Map
               column_order: 8
               size: 255
               db_column_name: obs_creator_name
               scs_verbosity: 3
               description: Name of the creator of the observation
               utype: DataID.creatorDID
               principal: 0
               indexed: 0
               std: 1
            type: text
         obs_publisher_did: !Column
            options: !Map
               column_order: 9
               size: 255
               db_column_name: obs_publisher_did
               scs_verbosity: 3
               description: Dataset identifier given by the publisher
               utype: Curation.publisherDID
               principal: 0
               indexed: 0
               std: 1
            type: text
         access_url: !Column
            options: !Map
               column_order: 10
               size: 255
               db_column_name: access_url
               scs_verbosity: 2
               description: URL used to access (download) dataset
               utype: Access.reference
               principal: 0
               indexed: 0
               std: 1
            type: text
         access_format: !Column
            options: !Map
               column_order: 11
               size: 50
               db_column_name: access_format
               scs_verbosity: 3
               description: File content format
               utype: Access.format
               principal: 0
               indexed: 0
               std: 1
            type: text
         access_estsize: !Column
            options: !Map
               column_order: 12
               size: 8
               db_column_name: access_estsize
               scs_verbosity: 3
               description: Estimated size of dataset in kilobytes
               utype: Dataset.Length
               principal: 0
               indexed: 0
               std: 1
            type: bigint
         s_ra: !Column
            options: !Map
               column_order: 13
               size: 8
               db_column_name: s_ra
               scs_verbosity: 1
               description: Central right ascension, ICRS
               utype: Char.SpatialAxis.Coverage.Location.Coord.Position2D.Value2.C1
               ucd: pos.eq.ra;meta.main
               principal: 1
               indexed: 0
               std: 1
            type: double precision
         s_dec: !Column
            options: !Map
               column_order: 14
               size: 8
               db_column_name: s_dec
               scs_verbosity: 1
               description: Central declination, ICRS
               utype: Char.SpatialAxis.Coverage.Location.Coord.Position2D.Value2.C2
               ucd: pos.eq.dec;meta.main
               principal: 1
               indexed: 0
               std: 1
            type: double precision
         s_fov: !Column
            options: !Map
               column_order: 15
               size: 4
               db_column_name: s_fov
               scs_verbosity: 3
               description: Diameter (bounds) of the covered region
               utype: Char.SpatialAxis.Coverage.Bounds.Extent.diameter
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         s_region: !Column
            options: !Map
               column_order: 16
               size: 50
               db_column_name: s_region
               scs_verbosity: 3
               description: Sky region covered by the data product (expressed in ICRS frame)
               utype: Char.SpatialAxis.Coverage.Support.Area
               principal: 0
               indexed: 0
               std: 1
            type: spoly
         s_resolution: !Column
            options: !Map
               column_order: 17
               size: 50
               db_column_name: s_resolution
               scs_verbosity: 3
               description: Spatial resolution of data as FWHM
               utype: Char.SpatialAxis.Resolution.refval.value
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         s_xel1: !Column
            options: !Map
               column_order: 18
               size: 4
               db_column_name: s_xel1
               scs_verbosity: 3
               description: Number of elements along the first spatial axis
               principal: 0
               indexed: 0
               std: 1
            type: integer
         s_xel2: !Column
            options: !Map
               column_order: 19
               size: 4
               db_column_name: s_xel2
               scs_verbosity: 3
               description: Number of elements along the second spatial axis
               principal: 0
               indexed: 0
               std: 1
            type: integer
         t_xel: !Column
            options: !Map
               column_order: 20
               size: 4
               db_column_name: t_xel
               scs_verbosity: 3
               description: Number of elements along the time axis
               principal: 0
               indexed: 0
               std: 1
            type: integer
         t_min: !Column
            options: !Map
               column_order: 21
               size: 8
               db_column_name: t_min
               scs_verbosity: 3
               description: Start time in MJD
               utype: Char.TimeAxis.Coverage.Bounds.Limits.StartTime
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         t_max: !Column
            options: !Map
               column_order: 22
               size: 8
               db_column_name: t_max
               scs_verbosity: 3
               description: Stop time in MJD
               utype: Char.TimeAxis.Coverage.Bounds.Limits.StopTime
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         t_exptime: !Column
            options: !Map
               column_order: 23
               size: 8
               db_column_name: t_exptime
               scs_verbosity: 3
               description: Total exposure time
               utype: Char.TimeAxis.Coverage.Support.Extent
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         t_resolution: !Column
            options: !Map
               column_order: 24
               size: 8
               db_column_name: t_resolution
               scs_verbosity: 3
               description: Temporal resolution FWHM
               utype: Char.TimeAxis.Resolution.Refval.valueResolution.Refval.value
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         em_xel: !Column
            options: !Map
               column_order: 25
               size: 4
               db_column_name: em_xel
               scs_verbosity: 3
               description: Number of elements along the spectral axis
               principal: 0
               indexed: 0
               std: 1
            type: integer
         em_min: !Column
            options: !Map
               column_order: 26
               size: 8
               db_column_name: em_min
               scs_verbosity: 3
               description: Start in spectral coordinates
               utype: Char.SpectralAxis.Coverage.Bounds.Limits.LoLimit
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         em_max: !Column
            options: !Map
               column_order: 27
               size: 8
               db_column_name: em_max
               scs_verbosity: 3
               description: Stop in spectral coordinates
               utype: Char.SpectralAxis.Coverage.Bounds.Limits.HiLimit
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         em_res_power: !Column
            options: !Map
               column_order: 28
               size: 8
               db_column_name: em_res_power
               scs_verbosity: 3
               description: Spectral resolving power
               utype: Char.SpectralAxis.Resolution.ResolPower.refVal
               principal: 0
               indexed: 0
               std: 1
            type: double precision
         o_ucd: !Column
            options: !Map
               column_order: 29
               size: 50
               db_column_name: o_ucd
               scs_verbosity: 3
               description: UCD of observable
               utype: Char.ObservableAxis.ucd
               principal: 0
               indexed: 0
               std: 1
            type: text
         pol_xel: !Column
            options: !Map
               column_order: 30
               size: 4
               db_column_name: pol_xel
               scs_verbosity: 3
               description: Number of polarization samples
               utype: Char.PolarizationAxis.stateList
               principal: 0
               indexed: 0
               std: 1
            type: integer
         pol_states: !Column
            options: !Map
               column_order: 31
               size: 10
               db_column_name: pol_states
               scs_verbosity: 3
               description: List of polarization states or NULL if not applicable
               utype: Char.PolarizationAxis.stateList
               principal: 0
               indexed: 0
               std: 1
            type: text
         instrument_name: !Column
            options: !Map
               column_order: 32
               size: 50
               db_column_name: instrument_name
               scs_verbosity: 3
               description: 
               utype: Provenance.ObsConfig.Instrument.name
               principal: 0
               indexed: 0
               std: 1
            type: text
      options: !Map
         description: IVOA Obscore
         description_long: This is the standard table for all TAP services
         tap.schema.name: ivoa
         scs_enabled: 1
         release_required: 0
         tap.table.name: obscore
```

**Comment by gsleap** [comment]
Oh and I've also got the logs that show the full stack trace too, if that helps. (See attached).
[casda_vo_tools.log](https://github.com/csiro-rds/casda_vo_tools/files/4441882/casda_vo_tools.log)

**Comment by gsleap** [comment]
Hi James,  Just confirming that your fix provided in email: "It looks like I made a mistake when adding a new field to the tap_configuration.yaml file. I am currently testing out the fix, however if you want to fix it quickly, you can change the column_order value from 11 to 12 on line 250 of tap_configuration.yaml in the block for the release_date column. Then restart the app and it should pick up the change and hopefully work." work perfectly, thanks. Once the fix is implemented, feel free to close the issue. For now I have fixed it my local repo only.

**Comment by jd-au** [comment]
We resolved this bug in CASDA v1.12, as Greg has described above.
